src/generateTasks.js

```diff
@@ -1,26 +1,23 @@
 'use strict'
 
+const path = require('path')
 const minimatch = require('minimatch')
-
-const readConfigOption = require('./readConfigOption')
+const getConfig = require('./getConfig')
+const resolveGitDir = require('./resolveGitDir')
 
 module.exports = function generateTasks(config, files) {
-  const linters = config.linters !== undefined ? config.linters : config
-  const resolve = file => files[file]
+  const normalizedConfig = getConfig(config) // Ensure we have a normalized config
+  const linters = normalizedConfig.linters
+  const gitDir = normalizedConfig.gitDir
+  const globOptions = normalizedConfig.globOptions
   return Object.keys(linters).map(pattern => {
     const commands = linters[pattern]
-    const globOptions = readConfigOption(config, 'globOptions', {})
-    const filter = minimatch.filter(
-      pattern,
-      Object.assign(
-        {
-          matchBase: true,
-          dot: true
-        },
-        globOptions
-      )
-    )
-    const fileList = Object.keys(files).filter(filter).map(resolve)
+    const filter = minimatch.filter(pattern, globOptions)
+    const fileList = files
+      // We want to filter before resolving paths
+      .filter(filter)
+      // Return absolute path after the filter is run
+      .map(file => path.resolve(resolveGitDir(gitDir), file))
```

Comment:
Can we avoid the repeated invocation of `resolveGitDir(gitDir)`?

Reply:
I assume the cost of calling is negligible so I don't see it as a problem. It helps with tests though.